backend/internal/service/store/user.go

```diff
@@ -89,3 +89,81 @@ func (mh *mongoHandler) SetWalletBalance(userName string, newBalance int) error
 	}
 	return nil
 }
+
+// Not Tested.
+func (mh *mongoHandler) AddWalletTransaction(userName string, walletTxID string, stockID string, is_debit bool, amount int, timeStamp int64) error {
+	var walletTx models.WalletTransaction = models.WalletTransaction{
+		UserName:   userName,
+		WalletTxID: walletTxID,
+		StockID:    stockID,
+		Is_debit:   is_debit,
+		Amount:     amount,
+		TimeStamp:  timeStamp,
+	}
+
+	// Add to 'wallet_transactions' collection
+	collection := mh.client.Database("day-trading-app").Collection("wallet_transactions")
+	_, err := collection.InsertOne(context.Background(), bson.M{"stock_tx_id": walletTx})
+	if err != nil {
+		return err
+	}
+
+	// * Add to user's entry in 'users' collection *
+
+	// Access the collection where user data is stored
+	collection = mh.client.Database("day-trading-app").Collection("users")
+
+	// Find the user by their username
+	var user models.User
+	err = collection.FindOne(context.Background(), bson.M{"user_name": userName}).Decode(&user)
+	if err != nil {
+		return err
+	}
+
+	user.WalletTxns = append(user.WalletTxns, walletTx)
+
+	// Add to the user's wallet transactions
+	_, err = collection.ReplaceOne(context.Background(), bson.M{"user_name": userName}, user)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Not Tested.
+func (mh *mongoHandler) DeleteWalletTransaction(userName string, walletTxID string) error {
+	// Remove from 'wallet_transactions' collection
+	collection := mh.client.Database("day-trading-app").Collection("wallet_transactions")
+	_, err := collection.DeleteOne(context.Background(), bson.M{"stock_tx_id": walletTxID})
+	if err != nil {
+		return err
+	}
+
+	// Remove from user's entry in 'users' collection
+
+	// Access the collection where user data is stored
+	collection = mh.client.Database("day-trading-app").Collection("users")
+
+	// Find the user by their username
+	var user models.User
+	err = collection.FindOne(context.Background(), bson.M{"user_name": userName}).Decode(&user)
+	if err != nil {
+		return err
+	}
+
+	// Remove the transaction
+	for i, tx := range user.WalletTxns {
+		if tx.WalletTxID == walletTxID {
+			user.WalletTxns = append(user.WalletTxns[:i], user.WalletTxns[i+1:]...)
```

Comment:
If you call UpdateStockOrderStatus, the transaction does not get added to the wallet transaction until the the stock transaction's order status gets updated to "Complete" this way you don't have to delete ever.